flake.nix

```diff
@@ -1,330 +1,79 @@
-        settings = {
-          ormolu.defaultExtensions = [
-            "TypeApplications"
-            "PatternSynonyms"
-          ];
-        };
-
-        hooks = {
-          nixpkgs-fmt.enable = true;
-          cabal-fmt.enable = true;
-          fourmolu.enable = true;
-          hlint.enable = true;
-        };
-      };
-
+  outputs = { self, nixpkgs, flake-utils, haskellNix, CHaP, pre-commit-hooks }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" "x86_64-darwin" ] (system:
+      let
+        pre-commit-check = pre-commit-hooks.lib.${system}.run {
+          src = ./.;
+          settings = {
+            ormolu.defaultExtensions = [
+              "TypeApplications"
+              "PatternSynonyms"
+            ];
+          };
 
-      mkDevEnv = system:
-        # Generic environment bringing generic utilities. To be used only as a
-        # shell. Include as a dependency to other shells to have the same
-        # utilities in the shell.
-        let
-          pkgs = nixpkgsFor system;
-          pkgs' = nixpkgsFor' system;
-        in
-        pkgs.stdenv.mkDerivation {
-          name = "Standard-Dev-Environment-with-Utils";
-          buildInputs = [
-            pkgs'.bashInteractive
-            pkgs'.cabal-install
-            pkgs'.fd
-            pkgs'.git
-            pkgs'.gnumake
-            pkgs'.haskellPackages.apply-refact
-            pkgs'.haskellPackages.cabal-fmt
-            pkgs'.haskellPackages.fourmolu
-            pkgs'.hlint
-            pkgs'.nixpkgs-fmt
-          ];
-          shellHook = (pre-commit-check-for system).shellHook +
-            ''
-              echo $name
-            '';
+          hooks = {
+            nixpkgs-fmt.enable = true;
+            cabal-fmt.enable = true;
+            fourmolu.enable = true;
+            hlint.enable = true;
+          };
         };
-
-      # Ply core
-      ply-core = rec {
-        ghcVersion = "8107";
-        compiler-nix-name = "ghc${ghcVersion}";
-
-        projectFor = system:
-          let
-            pkgs = nixpkgsFor system;
-            pkgs' = nixpkgsFor' system;
-            stdDevEnv = mkDevEnv system;
-            h = myhackages system compiler-nix-name;
-          in
-          (nixpkgsFor system).haskell-nix.cabalProject' {
-            inherit compiler-nix-name;
-            inherit (h) extra-hackages extra-hackage-tarballs;
-            src = ./.;
-            cabalProjectFileName = "cabal.project.core";
-            cabalProjectLocal = ''
-              allow-newer: size-based:template-haskell
-            '';
-            modules = [
-              ({ pkgs, ... }:
-                {
-                  packages = {
-                    cardano-crypto-praos.components.library.pkgconfig = pkgs.lib.mkForce [ [ pkgs.libsodium-vrf ] ];
-                    cardano-crypto-class.components.library.pkgconfig = pkgs.lib.mkForce [ [ pkgs.libsodium-vrf ] ];
+        overlays = [
+          haskellNix.overlay
+          (final: prev: {
```

Comment:
you don't really have to use an overlay here, but it's fine, I guess. Generally I prefer to not use overlays if it's possible because it sometimes makes caching worse.

I think what you want to do is to still declare the ply project as before and then just use the overlaid pkgs.